### PR TITLE
Improve regex toolchain docu

### PR DESCRIPTION
**Pull request by reckseba**
Does this help?

### CODE REVIEW

File: backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReferenceService.java [hunk 1]
```diff
@@ -127,6 +127,8 @@ private List<MatchInfo> findReferences(final String text) {
 
   record MatchInfo(String reference, int start, int end) {}
 
+  // This regex was crafted by a regex tool. Check /regex/README.md before editing. We need to keep
+  // both in sync.
```
Comment (hannes-mk) [inline backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReferenceService.java]: What does "both" refer to?

File: regex/README.md [hunk 2]
```diff
@@ -1,14 +1,35 @@
-# regex toolchain
+# Regular Expression Toolchain for Reference Detection
+## Motivation and Context
+When displaying an amending law, potential references (AV) should be highlighted and available for interaction. Documentalists could highlight them manually, but we'd like to support them with a semi-automated reference detection.
```
Comment (hannes-mk) [inline regex/README.md]: what's "AV"?

**Comment by hannes-mk** [comment]
Maybe not for the documentation, but for me: Where is that regex used, eventually? 
In the frontend or backend? If only one of them: why is the "regex" machinery outside that folder?

**Comment by hannes-mk** [comment]
Another thought: the regex is not tested, is it?

**Review by hannes-mk — APPROVED** [review]
I think your information is already better that what we have right now.

My other comments are meant as input to potentially making it even more informative.

File: regex/README.md (hunk 2 above)
Comment (reckseba) [reply regex/README.md]: Aktivverweis. But yes, that comes out of nowhere

**Comment by reckseba** [comment]
> Maybe not for the documentation, but for me: Where is that regex used, eventually? In the frontend or backend? If only one of them: why is the "regex" machinery outside that folder?

In the backend in the ReferenceService. I thought it might be obvious if I reference the service like that. I can extend it to make it more clear.

> Another thought: the regex is not tested, is it?

Whats a tested regex for you? In a way it's tested as every text file represents a bucket which is tested against the according one.

**Comment by hannes-mk** [comment]
I'll take the testing question to our dev weekly.

File: regex/README.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ When a certain case shall be captured, that is not yet handled, the idea is:
 The bitter pill of the current approach is, that one needs to check if the regex in Java code was altered in the meantime. Meaning if someone touched the regex there without changing it here, the next time a rendered regex will be inserted there, the changes will be overridden.
```
Comment (hannes-mk) [inline regex/README.md]:
```suggestion
The bitter pill of the current approach is, that one needs to check if the regex in Java code was altered in the meantime. Meaning if someone touched the regex there without changing it here, the next time a rendered regex will be inserted there, the changes will be overwritten.
```

**Pull request merged**